Composer/packages/client/src/pages/language-generation/index.tsx

```diff
@@ -37,12 +38,24 @@ const LGPage: React.FC<RouteComponentProps> = props => {
   const { lgFiles, dialogs } = state;
   const [editMode, setEditMode] = useState(false);
   const [fileValid, setFileValid] = useState(true);
-  const [codeRange, setCodeRange] = useState<CodeRange | null>(null);
+  const [codeRange, setCodeRange] = useState<CodeRange | undefined>(undefined);
+  const [lgEditor, setLgEditor] = useState<editor.IStandaloneCodeEditor | null>(null);
+  // const editorRef: MutableRefObject<editor.IStandaloneCodeEditor | null> = useRef(null);
 
+  const hash = props.location ? props.location.hash : '';
   const subPath = props['*'];
   const isRoot = subPath === '';
   const activeDialog = dialogs.find(item => item.id === subPath);
 
+  useEffect(() => {
+    if (hash && lgEditor) {
+      const matched = hash.match(/line=(\d+)/g);
```

Comment:
Prefer this approach.

```suggestion
      const match = /line=(\d+)/g.exec(hash);
```

Then you can look up `match[1]`, it will be the number.

If it is no-match, `match` will be falsy.

So the final code looks like:

```js
      const match = /line=(\d+)/g.exec(hash);

      if (match) {
        lgEditor.revealLine(+match[1]);
      }
```

Also, consider using `+` instead of `parseInt` or `Number()`. All of them are different, but `+` is fastest and work in you case. Some of them deal with string, undefined, etc, differently.